src/main/java/sorald/sonar/BestFitScanner.java

```diff
@@ -167,6 +171,45 @@ private boolean canRepair(E element) {
         }
     }
 
+    /**
+     * Post filter for position-matched candidates for elements that require special handling, such
+     * as jointly declared variables (i.e. multiple declarations in one statement).
+     */
+    private boolean candidatePostFilter(E element, RuleViolation violation) {
+        if (element instanceof CtVariable && ((CtVariable<?>) element).isPartOfJointDeclaration()) {
+            String ident =
+                    getPrecedingSymbol(violation, element.getPosition().getCompilationUnit());
+            return ident.equals(((CtVariable<?>) element).getSimpleName());
+        } else {
+            return true;
+        }
+    }
+
+    /**
+     * Get the first non-whitespace symbol preceding the source position of the given rule
+     * violation.
+     */
+    private static String getPrecedingSymbol(RuleViolation violation, CtCompilationUnit cu) {
+        String cuSource = cu.getOriginalSourceCode();
+        int searchStartPos =
+                calculateSourcePos(
+                        violation.getStartLine(),
+                        violation.getStartCol(),
+                        cu.getLineSeparatorPositions());
+        int symbolEndPos = reverseSearch(cuSource, searchStartPos, Character::isJavaIdentifierPart);
```

Comment:
Maybe naming these `identifierStartPos` and `identifierEndPos` would be more intuitive?

Reply:
:+1: